Zend/zend_compile.c

```diff
@@ -2690,6 +2798,16 @@ static zend_op *zend_delayed_compile_prop(znode *result, zend_ast *ast, uint32_t
 		opline = zend_delayed_compile_var(&obj_node, obj_ast, type, 0);
 		zend_separate_if_call_and_write(&obj_node, obj_ast, type);
 	}
+
+	if (nullsafe) {
+		uint32_t jmp_null_opnum = get_next_op_number();
+		zend_emit_op(NULL, ZEND_JMP_NULL, &obj_node, NULL);
+
+		zend_short_circuiting_scope *short_circuiting_scope = zend_current_short_circuiting_scope();
+		zend_stack *jmp_null_opnums = &short_circuiting_scope->jmp_null_opnums;
+		zend_stack_push(jmp_null_opnums, &jmp_null_opnum);
+	}
+
 	zend_compile_expr(&prop_node, prop_ast);
```

Comment:
Does this need similar handling to the compile_dim?

Reply:
Only if we implemented `?[]` which the RFC currently doesn't have.

Reply:
To be clear, what I have in mind is this:

```
(new stdClass)->{null?->x}->y;
```

This actually leaks right now, not sure if that's related though.

Reply:
> This actually leaks right now, not sure if that's related though.

Oh, I see. I'll check it out! Thanks

Reply:
That was actually a bug, the RHS of a property fetch also needs to create a new chain. I'll update that in the RFC too. Good catch!